### PR TITLE
Indirection avoiding trouble with turbojpeg and tiff.

### PR DESCRIPTION
**Pull request by cmangeat**
Raster rendering bug, trouble with jai native, jai ext, jpeg turbo and alpha channel removed via jai BandSelect operation.

### CODE REVIEW

File: src/extension/libjpeg-turbo/pom.xml [hunk 1]
```diff
@@ -61,4 +61,47 @@
...
+                -Dorg.geotools.image.test.enabled=${image.tests}
+                -Dorg.geotools.image.test.interactive=${interactive.image} -Duser.timezone=${user.timezone}
+                -Dwindows.leniency=${windows.leniency} -XX:+TieredCompilation -XX:TieredStopAtLevel=1
+              </argLine>
+              <enableAssertions>true</enableAssertions>
+              <printSummary>true</printSummary>
+              <testFailureIgnore>${allow.test.failure.ignore}</testFailureIgnore>
+              <excludedGroups>${test.excludedGroups}</excludedGroups>
+              <additionalClasspathElements>
+                <additionalClasspathElement>/usr/share/java/mlibwrapper_jai.jar</additionalClasspathElement>
```
Comment (aaime) [inline src/extension/libjpeg-turbo/pom.xml]: This assume a specific *nix filesystem, what happens if one tries to run it on windows?

File: src/extension/libjpeg-turbo/src/main/java/org/geoserver/map/turbojpeg/TurboJpegImageWorker.java [hunk 2]
```diff
@@ -103,7 +104,7 @@ public final void writeTurboJPEG(final OutputStream destination, final float com
         final int numBands = image.getSampleModel().getNumBands();
         if (hasAlpha) {
             final int requestedBands = numBands - 1;
-            if (ImageUtilities.isMediaLibAvailable()) {
+            if (ImageUtilities.isMediaLibAvailable() && !JAIExt.isJAIExtOperation("BandSelect")) {
```
Comment (aaime) [inline src/extension/libjpeg-turbo/src/main/java/org/geoserver/map/turbojpeg/TurboJpegImageWorker.java]: @dromagnoli what was the reason for the medialib check?

**Comment by cmangeat** [comment]
Goog morning,

Thank you for the time spent. I plan to have a scrupulous look at the end
of this a week (rebase, think about file layout, modify the test, maybe
open a ticket where "client value" could be discussed).

Best regards.

**Comment by cmangeat** [comment]
replaced by "libJegTurbo, jai native (mlib) and jai-ext, seems there is a trouble when removing alpha with… #2309"

**Pull request closed without merging**